### PR TITLE
[NPU] Fail for non continuous tensors

### PR DESCRIPTION
**Pull request by pereanub**
### Details:
 - *Fail if non continuous tensor is set*

### CODE REVIEW

**Review by sbutnari — APPROVED** [review]
new tests, yay!

File: src/plugins/intel_npu/tests/functional/behavior/infer_request_run.hpp [hunk 1]
```diff
@@ -766,6 +767,22 @@ TEST_P(BatchingRunSeqTests, CheckMultipleBatchingRunsSeq) {
     }
 }
 
+using ROITensorInference = OVInferRequestInferenceTests;
+
+TEST_P(ROITensorInference, InferenceROITensor) {
+    auto shape_size = ov::shape_size(m_param.m_shape);
+    auto model = OVInferRequestInferenceTests::create_n_inputs(1, ov::element::f32, m_param.m_shape);
+    auto execNet = ie->compile_model(model, target_device);
```
Comment (MirceaDan99) [inline src/plugins/intel_npu/tests/functional/behavior/infer_request_run.hpp:775]: I think we should start naming `compiledModel` vars as `executable networks` are from the older API.

**Review by razvanapetroaie — APPROVED** [review]

File: src/plugins/intel_npu/tests/functional/behavior/infer_request_run.hpp (hunk 1 above)
Comment (pereanub) [reply src/plugins/intel_npu/tests/functional/behavior/infer_request_run.hpp]: thx for comment, done

**Review by StefaniaHergane — APPROVED** [review]

File: src/plugins/intel_npu/tests/functional/behavior/infer_request_run.hpp (hunk 1 above)
Comment (MirceaDan99) [reply src/plugins/intel_npu/tests/functional/behavior/infer_request_run.hpp]: Also please consider using the camel case naming for the newly changed `auto compiled_model` variable, I think it sticks to our convention.

Comment (pereanub) [reply src/plugins/intel_npu/tests/functional/behavior/infer_request_run.hpp]: where is this convention written or used? all tests from this file are using the snake method.

**Pull request merged**